### PR TITLE
Adding pallet xcm interface precompile

### PR DESCRIPTION
**Pull request by wuzhong-papermoon**
### Description

Adding XCM interface precompile

### Checklist

- [ ✔️] I have added a label to this PR 🏷️
- [ ✔️] I have run my changes through Grammarly
- [ ✔️] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [ ✔️] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects
- [ ✔️] If pages have been moved around, I have run the `move-pages.py` script to move the pages and update the image paths on the chinese repo
    - [ ✔️] After the script has been run, I have created an additional PR in `moonbeam-docs-cn`
- [ ✔️] If images have been added, I have run the `compress-images.py` script to compress the images.
- [ ✔️] If variables (in variables.yml) need to be updated (such as a name change), I have updated the `moonbeam-docs-cn` repo to use the new variables
- [ ✔️] If this page requires a disclaimer, I have added one

### Corresponding PRs

Please link to any corresponding PRs here.

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [ ] No additional PRs are required after the translations are done

#### Items to be Updated

Please list any of the items that will need to be added or deleted after the translations are done here.

### CODE REVIEW

**Comment by wuzhong-papermoon** [comment]
@eshaben Could you check if the file structure is correct?

File: builders/ethereum/precompiles/interoperability/.pages [hunk 1]
```diff
@@ -5,3 +5,4 @@ nav:
   - 'X-Tokens': '/builders/interoperability/xcm/xc20/send-xc20s/xtokens-precompile/'
   - 'XCM Transactor': '/builders/interoperability/xcm/remote-execution/substrate-calls/xcm-transactor-precompile/'
   - 'XCM Utilities': '/builders/interoperability/xcm/xcm-utils/'
+  - 'XCM interface': '/builders/precompiles/xcm-interface/'
```
Comment (themacexpert) [inline builders/ethereum/precompiles/interoperability/.pages:8]:
```suggestion
  - 'XCM Interface': '/builders/precompiles/xcm-interface/'
```

File: builders/precompiles/xcm-Interface.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+     {{networks.moonbase.precompiles.xcm_interface }}
+     ```
+
+--8<-- 'text/builders/ethereum/precompiles/security.md'
+
+## The XCM Solidity Interface {: #the-xcm-solidity-interface }
+
+The [`XCMInterface.sol`](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank} is a Solidity interface that allows developers to interact with the precompile's functions.
+
+??? code "XcmInterface.sol"
```
Comment (themacexpert) [inline builders/precompiles/xcm-Interface.md:44]:
```suggestion
??? code "XCMInterface.sol"
```

File: builders/precompiles/xcm-Interface.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+### Checking Prerequisites {: #checking-prerequisites }
+
+To follow along with this tutorial, you will need to have your preferred EVM wallet configured and an account funded with native tokens. You can add Moonbeam to MetaMask wallet following this guide: [Interacting with Moonbeam Using MetaMask](/tokens/connect/metamask/){target=\_blank}.
+
+### Remix Set Up {: #remix-set-up }
+
+You can interact with the XCM interface precompile using [Remix](https://remix.ethereum.org){target=\_blank}. To add the precompile to Remix, you will need to:
+
+1. Get a copy of [`XcmInterface.sol`](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}
```
Comment (themacexpert) [inline builders/precompiles/xcm-Interface.md:130]:
```suggestion
1. Get a copy of [`XCMInterface.sol`](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}
```

File: builders/precompiles/xcm-Interface.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+2. Paste the file contents into a Remix file named `XcmInterface.sol`
```
Comment (themacexpert) [inline builders/precompiles/xcm-Interface.md:131]:
```suggestion
2. Paste the file contents into a Remix file named `XCMInterface.sol`
```

File: builders/precompiles/xcm-Interface.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+### Access the Contract {: #access-the-contract }
+
+Instead of deploying the precompile, you will access the interface given the address of the precompiled contract:
+
+1. Click on the **Deploy and Run** tab directly below the **Compile** tab in Remix. Please note that the precompiled contracts are already accessible at their respective addresses. Therefore, there is no deployment step
+2. Make sure **Injected Provider - Metamask** is selected in the **ENVIRONMENT** dropdown. Once you select **Injected Provider - Metamask**, you may be prompted by MetaMask to connect your account to Remix if it's not already connected
+3. Make sure the correct account is displayed under **ACCOUNT**
+4. Ensure **XCM - XcmInterface.sol** is selected in the **CONTRACT** dropdown. Given that it is a precompiled contract, there is no deployment step. Instead, you are going to provide the address of the precompile in the **At Address** field
+5. Provide the address of the precompile: `{{networks.moonbeam.precompiles.xcmInterface}}` and click **At Address**
```
Comment (themacexpert) [inline builders/precompiles/xcm-Interface.md:152]: Currently showing as blank for me
Screenshot 2024-08-07 at 10 37 57 AM

File: builders/ethereum/precompiles/overview.md [hunk 6]
```diff
@@ -89,6 +91,7 @@ The precompiled contracts are categorized by address and based on the origin net
     |   [Precompile Registry](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/precompile-registry/PrecompileRegistry.sol){target=\_blank}   |             {{networks.moonriver.precompiles.registry}}              |
     |                          [GMP](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/gmp/Gmp.sol){target=\_blank}                           |                {{networks.moonriver.precompiles.gmp}}                |
     |    [XCM Transactor V3](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/xcm-transactor/src/v3/XcmTransactorV3.sol){target=\_blank}     |         {{networks.moonriver.precompiles.xcm_transactor_v3}}         |
+    |                               [XCM interface](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}                               |             {{networks.moonbeam.precompiles.xcm_interface}}              |
```
Comment (themacexpert) [inline builders/ethereum/precompiles/overview.md:94]:
```suggestion
    |                               [XCM Interface](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}                               |             {{networks.moonbeam.precompiles.xcm_interface}}              |
```

File: builders/ethereum/precompiles/overview.md [hunk 7]
```diff
@@ -118,4 +121,5 @@ The precompiled contracts are categorized by address and based on the origin net
     |   [Precompile Registry](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/precompile-registry/PrecompileRegistry.sol){target=\_blank}   |             {{networks.moonbase.precompiles.registry}}              |
     |                          [GMP](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/gmp/Gmp.sol){target=\_blank}                           |                {{networks.moonbase.precompiles.gmp}}                |
     |    [XCM Transactor V3](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/xcm-transactor/src/v3/XcmTransactorV3.sol){target=\_blank}     |         {{networks.moonbase.precompiles.xcm_transactor_v3}}         |
+    |                               [XCM interface](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}                               |             {{networks.moonbeam.precompiles.xcm_interface}}              |
```
Comment (themacexpert) [inline builders/ethereum/precompiles/overview.md:124]:
```suggestion
    |                               [XCM Interface](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}                               |             {{networks.moonbeam.precompiles.xcm_interface}}              |
```

File: builders/ethereum/precompiles/overview.md [hunk 8]
```diff
@@ -60,7 +60,9 @@ The precompiled contracts are categorized by address and based on the origin net
     |               [Precompile Registry](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/precompile-registry/PrecompileRegistry.sol){target=\_blank}               |             {{networks.moonbeam.precompiles.registry}}              |
     |                                      [GMP](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/gmp/Gmp.sol){target=\_blank}                                       |                {{networks.moonbeam.precompiles.gmp}}                |
     |                [XCM Transactor V3](https://github.com/moonbeam-foundation/moonbeam/blob/master/precompiles/xcm-transactor/src/v3/XcmTransactorV3.sol){target=\_blank}                 |         {{networks.moonbeam.precompiles.xcm_transactor_v3}}         |
+    |                               [XCM interface](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}                               |             {{networks.moonbeam.precompiles.xcm_interface}}              |
```
Comment (themacexpert) [inline builders/ethereum/precompiles/overview.md:63]:
```suggestion
    |                               [XCM Interface](https://github.com/Moonsong-Labs/moonkit/blob/main/precompiles/pallet-xcm/XcmInterface.sol){target=\_blank}                               |             {{networks.moonbeam.precompiles.xcm_interface}}              |
```

**Review by themacexpert — CHANGES_REQUESTED** [review]
Looks good just a couple things I found!

File: builders/precompiles/xcm-Interface.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+???+ function "**transferAssetsToPara20**(_paraId, beneficiary, assets, feeAssetItem, weight_) — sends assets to another EVM-compatible appchain using the underlying `transfer_assets()` transaction included in the XCM pallet module"
+
+    === "Parameters"
+
+        - `paraId` (uint32) - the destination's appchain ID
+        - `beneficiary` (address) - the ECDSA-type account in the destination chain that will receive the tokens
+        - `assets` (`AssetAddressInfo[]` memory) - an array of assets to send
+        - `feeAssetItem` (uint32) - the index of the asset that will be used to pay fees
+        - `weight` (Weight memory)- the maximum gas to use in the whole operation. Setting uint64::MAX to `refTime` acts in practice as *unlimited weight*
```
Comment (eshaben) [inline builders/precompiles/xcm-Interface.md:60]: This is inconsistent with the way we document types on Moonbeam right now. There are some incoming changes, but for now, I think just throw the types into italics (i.e., `paraId` _uint32_, the destination's appchain ID) and that will be a quick win that is close enough

**Review by eshaben — COMMENTED** [review]
Looks good! Just some comments on the images:

- The style guide mentions that arrows should not be used with steps. So you can remove the arrows anywhere you're using steps.
- Are the steps 24x24px? This is also outlined in the style guide if you need more info 
- On `xcm-interface-2.webp` the steps appear to be larger than they are on `xcm-interface-3.webp` - can you tell me what size the steps are on 3? I can help you make some decisions here because I realize the size might need to change since they're so close together

As for the structure and where to put things, let me get back to you on this. But it will definitely need to be moved to the Interoperability > XCM section. I'll let you know where exactly!

**Comment by wuzhong-papermoon** [comment]
> Looks good! Just some comments on the images:
> 
> * The style guide mentions that arrows should not be used with steps. So you can remove the arrows anywhere you're using steps.
> * Are the steps 24x24px? This is also outlined in the style guide if you need more info
> * On `xcm-interface-2.webp` the steps appear to be larger than they are on `xcm-interface-3.webp` - can you tell me what size the steps are on 3? I can help you make some decisions here because I realize the size might need to change since they're so close together
> 
> As for the structure and where to put things, let me get back to you on this. But it will definitely need to be moved to the Interoperability > XCM section. I'll let you know where exactly!

Ah sorry that I didn't use standard style for steps. It should be all set now. 24x24 might be alright, would you take a look at the new images? @eshaben

**Comment by eshaben** [comment]
@wuzhong-papermoon yes, images look better! thanks!

**Review by eshaben — APPROVED** [review]
LGTM! @themacexpert just up to you to review/approve now 🙂

**Review by themacexpert — APPROVED** [review]

**Pull request merged**